engine/http.go

```diff
@@ -38,5 +40,20 @@ func (eng *Engine) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// can only be sent before the body.
 	// (we could possibly use http footers via chunked encoding, but I couldn't find
 	// how to use them in net/http)
+
+	// Cancel the job if the HTTP connection is closed.
+	if closeNotifier, ok := w.(http.CloseNotifier); ok {
```

Comment:
Hmm. Do we know the circumstances under which Engine.ServeHTTP is used? I put it in there because I saw that it was creating a job, so it stood to reason that it must also watch the `CloseNotifier`.